honeybadgermpc/test_batch_reconstruction.py

```diff
@@ -0,0 +1,199 @@
+import asyncio
+import random
+# from honeybadgermpc.wb_interpolate import decoding_message_with_none_elements
+from honeybadgermpc.field import GF
+from honeybadgermpc.polynomial import polynomialsOver
+from honeybadgermpc.batch_reconstruction import batch_reconstruction
+
+order = q = 17
+
+
+def random_element():
+    return random.randint(0, order)
+
+
+def simple_router(N):
```

Comment:
We should import this simple_router from somewhere rather than copy-pasting it.

Reply:
We could probably put it in a `pytest` fixture.

Reply:
I would suggest the following:

1. Move the test module `test_batch_reconstruction.py` under the `tests` dir.
2. Add a `simple_router` fixture in `tests/conftest.py`, e.g.:

```python
# tests/conftest.py
import asyncio
import random

from pytest import fixture


@fixture
def simple_router():

    def _simple_router(N):
        """
        Builds a set of connected channels
        @return (receives, sends)
        """
        # Create a mailbox for each party
        mbox = [asyncio.Queue() for _ in range(N)]

        def makeSend(i):
            def _send(j, o):
                # print('SEND %8s [%2d -> %2d]' % (o[0], i, j))
                # random delay
                asyncio.get_event_loop().call_later(
                    random.random()*1, mbox[j].put_nowait, (i, o))
            return _send

        def makeRecv(j):
            async def _recv():
                (i, o) = await mbox[j].get()
                # print('RECV %8s [%2d -> %2d]' % (o[0], i, j))
                return (i, o)
            return _recv

        sends = {}
        receives = {}
        for i in range(N):
            sends[i] = makeSend(i)
            receives[i] = makeRecv(i)
        return (sends, receives)

    return _simple_router
```
3. Remove the `simple_router` function definition in `tests/test_batch_reconstruction.py` and  use the `simple_router` fixture (defined in `conftest.py`) in the test like so:

```python
def test(simple_router):
    N = 4
    p = 73
    t = 1

    async def _test():
        # Test with simple case: n = 4, t =1
        # After AVSS, poly1 = x + 2, poly2 = 3x + 4, secret1 = 2, secret2 = 4
        # Hard code the shared secret value as input into batch_reconstruction function
        # The final constructed polynomial should be p = 4x + 2
        sends, recvs = simple_router(N)
        towait = []
        for i in range(N):
            # ...
```

Reply:
The above is demonstrated in 19737f7fc2a4728ea1bde9c1b185154de52fb1b5 (and 0768e061398cbaa58920d99290c55f01251739a3).